source/src/ModuleApplication.cc

```diff
@@ -225,32 +253,45 @@ namespace dqm4hep {
     //-------------------------------------------------------------------------------------------------
     
     void ModuleApplication::onStart() {
-      // get run control status in case it is already running
-      sendRequest(OnlineRoutes::RunControl::status(m_runControl.name()), net::Buffer(), [this](const net::Buffer &response){
-        core::json statusJson = core::json::parse(response.begin(), response.end());
-        bool rcRunning = statusJson.value<bool>("running", false);
-        if(rcRunning) {
-          core::json runJson = statusJson.value<core::json>("run", core::json({}));
-          core::Run run;
-          run.fromJson(runJson);
-          m_runControl.startNewRun(run);
-        }
-      });
-      if(ModuleApplication::STANDALONE == m_mode) {
+      if(ONLINE == appRunningMode()) {
+        // get run control status in case it is already running
+        sendRequest(OnlineRoutes::RunControl::status(m_runControl.name()), net::Buffer(), [this](const net::Buffer &response){
+          core::json statusJson = core::json::parse(response.begin(), response.end());
+          bool rcRunning = statusJson.value<bool>("running", false);
+          if(rcRunning) {
+            core::json runJson = statusJson.value<core::json>("run", core::json({}));
+            core::Run run;
+            run.fromJson(runJson);
+            m_runControl.startNewRun(run);
+          }
+        });        
+      }
+      if(STANDALONE == appModuleType()) {
         m_module->startOfCycle();
         m_cycle.startCycle(true);
+        m_standaloneTimer->start();
       }
-      else if(ModuleApplication::ANALYSIS == m_mode and m_runControl.isRunning()) {
+      else if(ANALYSIS == appModuleType() and m_runControl.isRunning()) {
         m_module->startOfCycle();
         m_cycle.startCycle(true);
       }
-      m_standaloneTimer->start();
+      if(EVENT_READER == appRunningMode()) {
+        core::Run run;
+        THROW_RESULT_IF(core::STATUS_CODE_SUCCESS, !=, m_eventReader->runInfo(run));
+        m_runControl.startNewRun(run);
+        m_module->startOfCycle();
+        m_cycle.startCycle();
```

Comment:
What's the difference between `startCycle()` and `startCycle(true)` ? (I'll can look it up in the doc later ;))

Reply:
If a cycle is running, it will stop the current one. The boolean flag decide whether an end of cycle has to be posted if the current cycle is stopped. I should probably check the consistency of every call here... Thanks !

Reply:
Ok, I've looked at the implementation of [`startCycle`](https://github.com/DQM4HEP/dqm4hep-online/blob/da248856b71092a86e3c73ba5aa9b6b875fc1204/source/src/Cycle.cc#L117) and I'm confused by the boolean parameter. 
If a cycle is already running and you set the bool to `true` then I agree you send an [`endOfCycle`](https://github.com/DQM4HEP/dqm4hep-online/blob/da248856b71092a86e3c73ba5aa9b6b875fc1204/source/src/Cycle.cc#L171) signal that stop the ongoing cycle -> You get out of the while loop and all your cycle properties are reset, including the `m_startCycleFlag`-> you get "stuck" into the `if(not m_startCycleFlag.load())` condition.